src/librustc/infer/region_inference/mod.rs

```diff
@@ -1543,7 +1543,7 @@ impl<'a, 'gcx, 'tcx> GenericKind<'tcx> {
         match *self {
             GenericKind::Param(ref p) => p.to_ty(tcx),
             GenericKind::Projection(ref p) => tcx.mk_projection(
-                p.trait_ref.clone(), p.item_name(tcx)),
+                p.trait_ref(tcx).clone(), p.item_name(tcx)),
```

Comment:
The question is where `trait_def_id` comes from there - there should be an associated type DefId somewhere, and you can combine that with `trait_ref.substs`.